src/runDiff.ts

```diff
@@ -0,0 +1,223 @@
+    };
+
+    // Then create octokit with the token
+    this.octokit = github.getOctokit(this.inputs.githubToken);
+    this.context = github.context;
+    core.debug("Context:");
+    core.debug(JSON.stringify(this.context));
+  }
+
+  /**
+   * Runtime entrypoint. Query for the last successful ran (not reran) jobs prior to this job and
+   * return the content of the outputs JSON as an output of this job. Outputs of this job will have
+   * the same key/name as the strings defined in the `needs` configuration.
+   */
+  async run() {
+    // Get job information
+    const { job_id, html_url } = await this.getJobId();
+
+    // Run the diff
+    const { result_code, summary } = await this.runDiff();
+
+    // Prepare outputs
+    const outputs: ActionOutputs = {
+      html_url,
+      job_id: job_id.toString(),
+      result_code,
+      stack: this.inputs.stack,
+      summary
+    };
+
+    // Set action outputs
+    Object.entries(outputs).forEach(([key, value]) => {
+      core.setOutput(key, value);
+    });
+
+    // Write outputs to file
+    const outputPath = path.join(this.inputs.workingDirectory, this.inputs.outputFilename);
+    fs.writeFileSync(outputPath, JSON.stringify(outputs));
+
+    if(result_code === "1") {
+      core.setFailed(summary);
+    }
+  }
+
+  /**
+   * Retrieves the job ID and HTML URL for the current workflow job.
+   * Supports pagination when querying the GitHub API.
+   * 
+   * @param jobName - Name of the job to find
+   * @returns Promise containing the job ID and HTML URL
+   * @throws Error if the job cannot be found
+   */
+  async getJobId(): Promise<{ job_id: number; html_url: string }> {
+    let page = 1;
+
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      const response = await this.octokit.rest.actions.listJobsForWorkflowRun({
+        ...github.context.repo,
+        run_id: github.context.runId,
+        per_page: 100,
+        page
+      });
+
+      const job = response.data.jobs.find(j => j.name === this.inputs.jobName);
+      if (job) {
+        return {
+          job_id: job.id,
+          html_url: job.html_url || ""
+        };
+      }
+
+      if (response.data.jobs.length < 100) {
+        throw new Error(`Could not find job with name ${this.inputs.jobName}`);
+      }
+
+      page++;
+    }
+  }
+
+  /**
+   * Executes the CDKTF diff command and processes its output.
+   * Supports both real execution and test mode with stub output.
+   * 
+   * @param inputs - Validated action inputs
+   * @returns Promise containing result code and summary
+   * @throws Error if the diff execution fails unexpectedly
+   */
+  async runDiff(): Promise<{ result_code: ActionOutputs["result_code"]; summary: string }> {
+    const outputPath = path.join(process.env.TMPDIR || "/tmp", "cdktf-diff.txt");
+    let diffCommand = this.inputs.stubOutputFile ? 
+      `cat ${this.inputs.stubOutputFile}` :
```

Comment:
This also [piped the output of `cat` into perl](https://github.com/snapsheet/cdktf-diff/blob/main/action.yml#L136) (I think this was to get rid of special characters?). Is that happening elsewhere?